### PR TITLE
#11728 correct IfForeignDiscountsExist names, not just translations. Use isDiscountEditable to also drive order of discounts being applied

### PR DESCRIPTION
**Pull request by adi-stefan**
https://github.com/metasfresh/metasfresh/issues/11728

### CODE REVIEW

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/pricing/SubscriptionPricingRule.java [hunk 1]
```diff
@@ -119,9 +119,13 @@ private IPricingResult aggregateSubscriptionDiscount(
 	{
 		if (subscriptionDiscountLine != null
 				&& !subscriptionPricingResult.isDisallowDiscount()
-				&& (subscriptionDiscountLine.isPrioritiseOwnDiscount() || !subscriptionPricingResult.isDiscountCalculated()) )
+				&& (subscriptionDiscountLine.isPrioritiseOwnDiscount() || !subscriptionPricingResult.isDiscountCalculated()))
 		{
 			subscriptionPricingResult.setDiscount(subscriptionDiscountLine.getDiscount());
+			if (subscriptionDiscountLine.isPrioritiseOwnDiscount())
+			{
+				subscriptionPricingResult.setDiscountEditable(false);
```
Comment (metas-ts) [inline backend/de.metas.contracts/src/main/java/de/metas/contracts/pricing/SubscriptionPricingRule.java:127]: ok, so i guess that's the answer to my question above. 
but as i wrote, i think `IsDiscountEditable` is related to the user, not to following rules.
Therefore i think we need an additional property like `dontOverrideDiscountAdvice` or something

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/pricing/SubscriptionPricingRule.java [hunk 2]
```diff
@@ -228,7 +231,11 @@ private static void copyDiscountIntoResultIfAllowedByPricingContext(
 	{
 		if (!pricingCtx.isDisallowDiscount())
 		{
-			result.setDiscount(subscriptionPricingResult.getDiscount());
+			if (result.isDiscountEditable())
+			{
+				result.setDiscount(subscriptionPricingResult.getDiscount());
+				result.setDiscountEditable(subscriptionPricingResult.isDiscountEditable());
```
Comment (metas-ts) [inline backend/de.metas.contracts/src/main/java/de/metas/contracts/pricing/SubscriptionPricingRule.java:237]: i would say this `result.setDiscountEditable` still makes sense..and as a 3rd line, here we should also copy the `dontOverrideDiscountAdvice` or whatever property you end up with.

**Review by metas-ts — CHANGES_REQUESTED** [review]

File: backend/de.metas.business/src/main/java/de/metas/pricing/rules/Discount.java [hunk 3]
```diff
@@ -80,9 +80,9 @@ public boolean applies(final IPricingContext pricingCtx, final IPricingResult re
 			return false;
 		}
 
-		if (result.isDiscountCalculated())
+		if (!result.isDiscountEditable())
```
[the remark this answers is not in the document]
Comment (adi-stefan) [reply backend/de.metas.business/src/main/java/de/metas/pricing/rules/Discount.java]: Changed to `isDontOverrideDiscountAdvice`

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/pricing/SubscriptionPricingRule.java (hunk 1 above)
Comment (adi-stefan) [reply backend/de.metas.contracts/src/main/java/de/metas/contracts/pricing/SubscriptionPricingRule.java]: added `isDontOverrideDiscountAdvice`

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/pricing/SubscriptionPricingRule.java (hunk 2 above)
Comment (adi-stefan) [reply backend/de.metas.contracts/src/main/java/de/metas/contracts/pricing/SubscriptionPricingRule.java:237]: added.

**Review by metas-ts — APPROVED** [review]

**Pull request merged**